changes/probe.py

```diff
@@ -58,11 +68,20 @@ def has_changelog():
 
 
 def has_readme():
-    """README.md"""
+    """README"""
+    readme = None
+    message = 'Create a README file'
+    for filename in os.listdir("."):
+        readme = re.match(r"README($|\.\w{2,})", filename, re.I)
```

Comment:
https://gist.github.com/goldsborough/03ce45bbc2128a85d3ce

See timing above. So, your `any` is most efficient. However, we'd have to do without the more detailed error message about there being a README, but with an invalid file extension, when that's the case. The error message could just state *Please add a README (with a valid file extension)*. Else, if we do want different error messages for when there is no README at all and when it simply has the wrong extension, see `exists2()` (most efficient I tried for differentiated error messages) in that gist: First do a glob for `README*`, see if any at all present (first case, no README), and if glob found files check if any with a valid extension is among them (else, second case, README with invalid extension).